cli/src/config.rs

```diff
@@ -202,6 +202,8 @@ fn create_dir_all(path: &Path) -> std::io::Result<()> {
 #[derive(Clone, Default, Debug)]
 struct UnresolvedConfigEnv {
     config_dir: Option<PathBuf>,
+    #[cfg(target_os = "macos")]
+    macos_legacy_config_dir: Option<PathBuf>,
```

Comment:
Make `config_dirs: Vec<PathBuf>` instead? Fewer platform-specific codes is better, and we wouldn't have to care about extra allocation cost.

Reply:
That was the original idea, but it really is only macOS, and this behaviour should really be deprecated and removed in a future release.

Reply:
Should we deprecate the current macOS config path? Since people appears to have different opinions, I think it's good to support both Unixy and macOS paths. There might be Windows guys who prefer XDG path, but I don't know.

Reply:
FWIW, I think that the `dirs` author is close to the only person who thinks that the current path is more correct for CLI programs (and it's based on a misleading of Apple's guidelines, e.g. the subdirectory is meant to be a bundle ID, which we don't have as we're not an app bundle, and it's for non-user-authored files managed by the app).

Reply:
Thanks. In that case, deprecating non-XDG path makes sense.